syntax = "proto3";

package protobuf;

message Authentication
{
    string name = 1;
    bytes password = 2;
}

message AuthenticationAnswer
{
    enum Result
    {
        success = 0;
        unknown = 1;
        wrongPassword = 2;
        other = 3;
    }
    Result result = 1;
}

message Achievement
{
    GameIdent gameID = 1;

    message Ident
    {
        uint64 ID = 1;

        enum Type
        {
            complete = 0;
            add = 1;
        }
        Type type = 2;
    }
    repeated Ident achievements = 2;
}

message GameIdent
{
    uint64 gameID = 1;
    bytes hash = 2;
}

